standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java

```diff
@@ -5088,10 +5089,8 @@ private boolean isDatabaseRemote(String name) {
 
   private void deleteParentRecursive(Path parent, int depth, boolean mustPurge, boolean needRecycle)
```

Comment:
Is there a reason why it can not use [HCFS API to delete dir recursively](https://hadoop.apache.org/docs/current/api/org/apache/hadoop/fs/FileSystem.html#delete-org.apache.hadoop.fs.Path-boolean-)?

Reply:
HCFS Api deletes all children recursively while this method (deleteParentRecursive) deletes a file and keeps on deleting parent directories if they are empty.

Reply:
I see. Did we consider just calling a non-recursive `deleteDir` on parent instead of checking if it's empty and based on the delete success/failure try to delete its parent recursively?

Reply:
I think it works if deleteDir fails when the dir is non-empty with no side effect. But we need to make sure all filesystems that Hive supports work in this way. Also I noticed that deleteDir calls moveToTrash first, so it could be more complex: https://github.com/apache/hive/blob/f2de30ca8bc2b63887496775f9a0769057a17ee0/standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreFsImpl.java#L41. Avoiding duplicated checks seems to be safer.